pkgs/development/python-modules/atheris/default.nix

```diff
@@ -0,0 +1,59 @@
+{
+  lib,
+  stdenv,
+  buildPythonPackage,
+  fetchFromGitHub,
+  llvmPackages,
+  pybind11,
+  pypaInstallHook,
+  python,
+  setuptoolsBuildHook,
+  pythonAtLeast,
+}:
+
+buildPythonPackage rec {
+  pname = "atheris";
+  version = "2.3.0";
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = "atheris";
+    rev = "refs/tags/${version}";
+    hash = "sha256-d5T+0YnHMS9nw8lyIS2TOQhwVJDc5dLp9tlpdoViPSs=";
+  };
+
+  postPatch = ''
+    patchShebangs setup_utils
+  '';
+
+  nativeBuildInputs = [
+    pybind11
+    pypaInstallHook
+    setuptoolsBuildHook
+  ];
+
+  pythonImportsCheck = [ "atheris" ];
+
+  env = {
+    LIBFUZZER_LIB = lib.concatStrings [
+      (lib.getLib llvmPackages.compiler-rt)
+      "/lib/"
+      (lib.toLower stdenv.hostPlatform.uname.system)
+      "/libclang_rt.fuzzer_no_main-"
+      stdenv.hostPlatform.uname.processor
+      ".a"
+    ];
+  };
+
+  meta = {
+    description = "Coverage-guided fuzzer for Python and Python extensions";
+    homepage = "https://github.com/google/atheris";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ getchoo ];
+    mainProgram = "atheris";
+    platforms = lib.intersectLists python.meta.platforms llvmPackages.compiler-rt.meta.platforms;
+    # https://github.com/google/atheris/issues/82
+    broken = pythonAtLeast "3.12";
```

Comment:
Why use `meta.broken` instead of `disabled`?

Reply:
As in, I don't know  which one is preferred in `python-modules` for such cases.

Reply:
AFAICT `disabled` is for cases where upstream has a minimal Python version requirement (e.g. `python3>=3.8`), but `broken` is for cases where upstream *intends* it to work for a version but it currently doesn't for any reason. In order words, `disabled` is for an intentional lack of support, while `broken` is more unintentional breakage.

Reply:
Yeah, that's what I'd guessed but good to see I'm not the only one who's _kind of_ uncertain.